src/sinks/prometheus/mod.rs

```diff
@@ -23,6 +24,14 @@ pub enum PrometheusRemoteWriteAuth {
         password: String,
     },
 
+    /// Bearer authentication.
+    ///
+    /// A bearer token (OAuth2, JWT, etc) is passed as-is.
+    Bearer {
```

Comment:
I'm wondering if there is any way we could embed the `Auth` enum in `src/http.rs` to avoid them falling out of sync.

Reply:
I actually went that route initially, but it then changes the config option for the auth to be nested if they want to use the `http::Auth` (basic/bearer).
This caused the unit tests to fail and I guess would be considered a breaking change?

I circle back to that though...

Reply:
Some serde flattening perhaps?

Reply:
Right, it'd need to be flattened. Unfortunately https://github.com/serde-rs/serde/issues/1402 makes it seem like it isn't possible to flatten enum options, but I might be missing something.

Reply:
Either way, happy to see this refactoring done as a second pass though.

Reply:
:eyes: looking into that now...

Reply:
So far I've not been able to get that to work, I'll open a follow-up issue.

Reply:
https://github.com/vectordotdev/vector/issues/15121